prae_macro/src/lib.rs

```diff
@@ -36,7 +36,7 @@ pub fn define(input: TokenStream) -> TokenStream {
         GuardClosure::Ensure(EnsureClosure(closure)) => quote! {
             fn validate(v: &Self::Target) -> Option<prae::ValidationError> {
                 let f: fn(&Self::Target) -> bool = #closure;
-                if f(v) { None } else { Some(prae::ValidationError) }
+                if f(v) { None } else { Some(prae::ValidationError::new::<#ty>("Validation failed.") ) }
```

Comment:
```suggestion
                if f(v) { None } else { Some(prae::ValidationError::new::<#ty>("provided value is invalid") ) }
```

Reply:
Error messages can be chained, so it's better to keep it lowercase and without ending dot

Reply:
Fair point